src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -228,16 +257,29 @@ export const FoxEthProvider = ({ children }: FoxEthProviderProps) => {
     lpAssetPrecision,
   ])
 
+  useEffect(() => {
+    if (!isMarketDataReady && ethMarketData.price && foxMarketData.price) setIsMarketDataReady(true)
+  }, [ethMarketData.price, foxMarketData.price, isMarketDataReady])
+
   // reload opportunities when wallet changes
   useEffect(() => {
-    if (!(connectedWalletEthAddress && lpApr)) return
+    if (
+      !(
+        connectedWalletEthAddress &&
+        lpApr &&
+        isMarketDataReady &&
+        lpAssetPrecision &&
+        foxAssetPrecision
+      )
+    )
+      return
     ;(async () => {
       fetchFarmingOpportunities()
       fetchLpOpportunity()
     })()
-    // to avoid refetching when the callback functions signatures change
+    // market data causes fetch callback function to create a new signature
```

Comment:
Just being a nitpick
```suggestion
    // market data causes fetch callback function to create a new reference
```